### PR TITLE
dd: rewrite it

### PR DESCRIPTION
**Pull request by rminnich**
I had this vision in the beginning of goroutines talking through
memory pipes and doing one transform at each step.

The juice is not worth the squeeze. We don't need all those
transforms; the main need is blocking.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

File: cmds/dd/dd.go [hunk 1]
```diff
@@ -48,58 +44,31 @@ var (
...
+		n, err := io.ReadFull(r, b)
+		if n == 0 || (err != nil && err != io.EOF) {
+			break
 		}
-		if bsc == 0 {
-			return
+		var out []byte
+		switch conv {
+		case "ucase":
+			out = []byte(strings.ToUpper(string(b)))
```
Comment (rjoleary) [inline cmds/dd/dd.go]: Golang has a `bytes.ToUpper` and `bytes.ToLower` for this.

**Comment by rminnich** [comment]
This is a snapshot, I'm done for now. Spent some time with what looks like a Go bug but worked around it anyway.

**Comment by rminnich** [comment]
OK I added a test and the conv= stuff works too. 

I'm going to push this in if it passes.

**Comment by rminnich** [comment]
The good news is that (here is a first) we're actually faster than the gnubin dd. It's never less than 20% faster.

**Pull request merged**